koku/api/query_params.py

```diff
@@ -425,25 +387,61 @@ def _set_access_ocp_all(self, provider, filter_key, access_key, raise_exception=
             elif access_list:
                 self.parameters["filter"][filter_key] = access_list
 
-    def _set_tag_keys(self):
+    def search_url_for_prefixes(self, prefix_list):
+        """Searches url and returns true if any prefix is found in a url."""
+        # the url_data property is percent encoded
+        # https://en.wikipedia.org/wiki/URL_encoding
+        # %5Btag == [tag:
+        # %5Bor%3Atag == [or:tag
+        decoded_url = unquote(self.url_data)
+        for prefix in prefix_list:
+            if f"[{prefix}" in decoded_url:
+                return True
+        return False
```

Comment:
An alternative solution. Not sure if it's any more readable, though.
```suggestion
        return any(f"[{prefix}" in decoded_url for prefix in prefix_list)
```